frame/executive/src/lib.rs

```diff
@@ -220,6 +220,25 @@ where
 		weight
 	}
 
+	/// Execute given block, but don't do any of the [`final_checks`].
+	///
+	/// Should only be used for testing.
+	#[cfg(feature = "try-runtime")]
+	pub fn execute_block_no_check(block: Block) -> frame_support::weights::Weight {
+		Self::initialize_block(block.header());
+		Self::initial_checks(&block);
+
+		let (header, extrinsics) = block.deconstruct();
+
+		Self::execute_extrinsics_with_book_keeping(extrinsics, *header.number());
+
+		// don't call `final_checks`, but do finalize the block. This is critical to clear transient
+		// storage items, such as weight.
+		let _header = <frame_system::Pallet<System>>::finalize();
+
```

Comment:
Would wonder if there is some migration that would pass all the tests written, but then actually cause a problem on final checks, but we dont see that now